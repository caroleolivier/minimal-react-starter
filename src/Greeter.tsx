import * as React from 'react';
import * as ReactDOM from 'react-dom';

interface GreeterProps {
    name: string;
}

export class Greeter extends React.Component<GreeterProps, {}> {
    render() {
        return <h1>Hello {this.props.name}</h1>;
    }
}