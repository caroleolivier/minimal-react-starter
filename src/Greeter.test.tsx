import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as testRenderer from 'react-test-renderer';

import { Greeter } from './Greeter';

describe('Greeter component', () => {
    let name: string ;
    let elt: React.ReactElement<Greeter>;

    beforeEach(() => {
        name = 'johndoe';
        elt = <Greeter name={name} />;
    });

    test('renders without crashing', () => {
        const div: HTMLDivElement = document.createElement('div');

        expect(() => ReactDOM.render(elt, div)).not.toThrow();
    });

    test('initial DOM matches snapshot', () => {
        const component = testRenderer.create(elt);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot('rendering');
    });
});