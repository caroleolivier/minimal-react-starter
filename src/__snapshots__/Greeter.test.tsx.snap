// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<h1>
  Hello 
  johndoe
</h1>
`;
